Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person/GetNearestDrivers.hs

```diff
@@ -110,3 +121,55 @@ getNearestDrivers cityServiceTiers serviceTiers fromLocLatLong radiusMeters merc
+      | haveScheduled && isRental -> return $ canTakeRental driverInfo.latestScheduledBooking now minimumScheduledBookingLeadTimeInSecs
+      | checkUsingTime driverInfo.latestScheduledBooking now scheduledRideFilterExclusionThresholdInSecs -> do
+        dropLoc <- currentSearchInfo.dropLocation & fromMaybeM (InternalError "Error: dropLocation not found in currentSearchInfo.")
+        scheduledPickup <- driverInfo.latestScheduledPickup & fromMaybeM (InternalError "Error: latestScheduledPickup not found in driverInfo.")
+        currentDroptoScheduledPickupDistance <-
+          TMaps.getDistance merchantId merchantOpCityId $
+            TMaps.GetDistanceReq
+              { origin = dropLoc,
+                destination = scheduledPickup,
+                travelMode = Just TMaps.CAR,
+                distanceUnit = Meter
+              }
+        routeDistance <- currentSearchInfo.routeDistance & fromMaybeM (InternalError "Error: routeDistance not found in currentSearchInfo.")
+        avgSpeeds <- transporterConfig.avgSpeedOfVehicle & fromMaybeM (InternalError "Error: avgSpeeds not found in transporterConfig.")
+        let avgSpeedOfSUVInKM = avgSpeeds.suv
+            destToPickupDistance = currentDroptoScheduledPickupDistance.distance
+            totalDistanceinM = routeDistance + destToPickupDistance
+            totalDistanceinKM = (fromIntegral (totalDistanceinM.getMeters) :: Double) / 1000
+            totalTimeinDoubleHr = (totalDistanceinKM / fromIntegral (avgSpeedOfSUVInKM.getKilometers)) :: Double
+            totalTimeInSeconds = realToFrac (totalTimeinDoubleHr * 3600) :: NominalDiffTime
+            expectedEndTime = addUTCTime totalTimeInSeconds now
+        let isRidePossible = case driverInfo.latestScheduledBooking of
+              Just latestScheduledBooking ->
+                let timeDifference = diffUTCTime latestScheduledBooking (addUTCTime transporterConfig.scheduleRideBufferTime expectedEndTime) -- can be configured later current: 30mins buffer time
+                 in timeDifference > 0
+              Nothing -> False
+        return isRidePossible
+      | otherwise -> return True
+  where
+    canTakeRental :: Maybe UTCTime -> UTCTime -> NominalDiffTime -> Bool
+    canTakeRental mbLatestScheduledBooking now minimumScheduledBookingLeadTimeInSecs =
+      case mbLatestScheduledBooking of
+        Nothing -> True
+        Just latestScheduledBooking ->
+          let timeDifference = diffUTCTime latestScheduledBooking now
+           in timeDifference > minimumScheduledBookingLeadTimeInSecs --can accept rental if time X is more than 24 hours
+    checkUsingTime :: Maybe UTCTime -> UTCTime -> NominalDiffTime -> Bool
+    checkUsingTime mbLatestScheduledBooking now seconds =
+      case mbLatestScheduledBooking of
+        Nothing -> False
+        Just latestScheduledBooking ->
+          let timeDifference = diffUTCTime latestScheduledBooking now
+           in timeDifference < seconds
```

Comment:
It should be less than or greater than?

Reply:
1. in canTakeRental we are just checking that timeDifference between current time and scheduledBookingStartTime should be more than 24 hours .
if timeDifference > 24 hours -> then it should return true else return false.

2.in checkUsingTime ->  timeDifference should be less than 7200 seconds (2 hours)